### ISSUE TITLE
STI filtration

### ISSUE THREAD

**Issue by Mark1Z** [mikro-orm/mikro-orm#1252]
**Describe the bug**
Child entities aren't filtrated

**To Reproduce**
1. Create entities:
```
@Entity({
  discriminatorColumn: 'type',
  discriminatorMap: { person: 'Person', employee: 'Employee' },
})
export abstract class BasePerson {
  @PrimaryKey()
  id!: number;

  @Enum()
  type!: 'person' | 'employee';
}

@Entity()
export class Person extends BasePerson {
  @PrimaryKey()
  id!: number;
}

@Entity()
export class Employee extends BasePerson {
  @PrimaryKey()
  id!: number;
}
```
2. Save some data:
```
const person = new Person();
const employee = new Employee();

this.em.persist(person);
this.em.persist(employee);

await this.em.flush();
```
3. Try to get only "Person" entities:
```
await this.em.getRepository(Person).findAll();
```
4. Get the result:
```
[{"id":1,"type":"person"},{"id":2,"type":"employee"}]
```

**Expected behavior**
Get filtrated "Person" entities.

**Versions**

| Dependency | Version |
| node | v12.20.0 |
| typescript | 4.0.5 |
| mikro-orm | 4.3.4 |
| mysql | 8.0.22 |

**Comment by thatsnotright** [comment]
Just upgraded to 4.4, I think this introduced an issue for our use case.  We have a query looking for all entities on an STI entity so we need to ignore the discriminator column value.  If I specify 'null' for the discriminatorValue column name it still sets this to our base entity.  I've been looking through the docs but I'm not seeing how I can use a typed repo and `await repo.findOneOrFail(` to get all entities which match the criteria, regardless their discriminatorValue.  

Is this case still possible?  I've tried setting the discriminatorColumn value to '%' and null in the filter for findOneOrFail but neither works, the queries generated are still setting to the base entity name.

Any hints?

**Comment by B4nan** [comment]
Sorry about that, already know about this issue from slack. You can work this around by querying the base type, that do not have `discriminatorValue` - but that value is autogerated, so you also need to specify `abstract: true` in the decorator so we know the `discriminatorValue` should not be there. 

Some docs about that are here: https://mikro-orm.io/docs/inheritance-mapping/#explicit-discriminator-column

Will be also shipping fix later this week, that will adjust the discriminator condition so that it is even possible to have the condition there (aka to have base type with `discriminatorValue`). Instead of strict equals check for single value it will use `where in ()` with list of all possible discr values based on the inheritance graph.

**Comment by B4nan** [comment]
@thatsnotright @Langstra the issue with child entities should be fixed in master (df298a1), will ship 4.4.1 probably tomorrow or the day after.

**Comment by Langstra** [comment]
Thanks! We already made our base entity abstract which did fix the problem for us, but I think this is a great solution for many of us as well.
